src/lightning_utilities/core/requirements.py

```diff
@@ -0,0 +1,112 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
Since this is for CI, it should not be in the core directory. We used to have a "dev" directory, but it was renamed to test. I suggest it's moved there or somewhere equivalent

Reply:
this is used in setup.py and CI benefits from it, but I would still consider is as core

Reply:
Why? Where would somebody use it in their source code? It's strictly for CI. The logic is very overfitted to our CI workflow, particularly the `adjust` logic

Reply:
`setup.py` is not CI :D but mandatory part of the package :otter:

Reply:
It's not, nowadays you can define a package configuration entirely in files like `setup.cfg` or `pyproject.toml`

Reply:
hmm so shall then remove this also from lightning? :thinking:

Reply:
What is your point? In lightning, this logic is not in the core

That's what I'm suggesting: don't put this in utilities core

Reply:
it is moved :)

Reply:
Didn't notice! Resolving